### PR TITLE
[GCP] [Billing] Improve documentation

### LINKED ISSUE #9914: Improve GCP Billing docs
We've noticed a gap in our GCP Billing integration docs — specifically, there's no clear explanation or example configurations for parameters like `dataset_id`. This can make setup confusing for users.

It'd be helpful to include a simple section with examples in our docs. For instance, we could explain that the `dataset_id` is the ID of your BigQuery dataset where billing data is stored, and `table_pattern` defines which tables to pull from. This would make setting up the integration way easier for everyone.

cc @ishleenk17

### PR DESCRIPTION
**Pull request by ghost**
## Overview

Improves GCP Billing documentation to clarify what each config param means.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #9914

### CODE REVIEW

File: packages/gcp/_dev/build/docs/billing.md [hunk 1]
```diff
@@ -12,6 +12,41 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
...
+
+The `table_pattern` defines which tables to pull from within the specified dataset. This can be set to either `gcp_billing_export_v1` for standard usage cost data or `gcp_billing_export_resource_v1` for detailed usage cost data.
+
+### cost_type
+
+The `cost_type` specifies the type of cost data you want to retrieve from your billing data.
+
+- `regular`: This cost type includes all the regular costs associated with your usage of GCP services. This does not include any taxes, adjustments, or rounding errors.
+
+- `tax`: This cost type includes all the taxes associated with your usage of GCP services. This does not include the regular costs, adjustments, or rounding errors.
```
Comment (shmsr) [inline packages/gcp/_dev/build/docs/billing.md:31]:
```suggestion
- `tax`: This cost type includes all the taxes associated with your usage of GCP services. This includes all the applicable taxes based on your billing location and the services used.
```

File: packages/gcp/_dev/build/docs/billing.md [hunk 2]
```diff
@@ -12,6 +12,41 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
...
+
+The `dataset_id` is the ID of your BigQuery dataset where your billing data is stored. This is a unique identifier for the dataset within the project. You can find this ID in your Google Cloud Console under BigQuery.
+
+### table_pattern
+
+The `table_pattern` defines which tables to pull from within the specified dataset. This can be set to either `gcp_billing_export_v1` for standard usage cost data or `gcp_billing_export_resource_v1` for detailed usage cost data.
+
+### cost_type
+
+The `cost_type` specifies the type of cost data you want to retrieve from your billing data.
```
Comment (shmsr) [inline packages/gcp/_dev/build/docs/billing.md:27]:
```suggestion
The `cost_type` parameter enables you to filter the cost data based on specific cost categories. You can select one of the following options:
```

Comment (shmsr) [reply packages/gcp/_dev/build/docs/billing.md:27]: Just checking. We do not allow multiple values here, right?

File: packages/gcp/_dev/build/docs/billing.md [hunk 3]
```diff
@@ -12,6 +12,41 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
...
+Here's an example of what your configuration might look like:
+
+```
+dataset_id: "my_billing_dataset"
+table_pattern: "gcp_billing_export_resource_v1"
+project_id: "my_project"
+cost_type: "regular"
+```
+
+In this example, the agent will pull data from the `gcp_billing_export_resource_v1` table within the `my_billing_dataset` dataset.
```
Comment (shmsr) [inline packages/gcp/_dev/build/docs/billing.md:48]:
```suggestion
In this example, the Agent will extract data from the `gcp_billing_export_resource_v1` table in the `my_billing_dataset` dataset.
```

File: packages/gcp/_dev/build/docs/billing.md [hunk 4]
```diff
@@ -12,6 +12,41 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
...
 
+## Configuration Parameters
+
+### dataset_id
+
+The `dataset_id` is the ID of your BigQuery dataset where your billing data is stored. This is a unique identifier for the dataset within the project. You can find this ID in your Google Cloud Console under BigQuery.
+
+### table_pattern
+
+The `table_pattern` defines which tables to pull from within the specified dataset. This can be set to either `gcp_billing_export_v1` for standard usage cost data or `gcp_billing_export_resource_v1` for detailed usage cost data.
```
[the remark this answers is not in the document]
Comment (shmsr) [reply packages/gcp/_dev/build/docs/billing.md:23]: Do you think adding `parameter` word makes it better? This is a nitpick. Feel free to decide. If yes, then please add this to other options too. Addition of the keyword `parameter` and changing the tone a little should do.

File: packages/gcp/_dev/build/docs/billing.md [hunk 5]
```diff
@@ -12,6 +12,41 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
 
 For detailed usage cost data, set the table pattern to `gcp_billing_export_resource_v1`. Detailed tables include the standard fields and additional fields, such as `effective_price`, enabling a more granular view of expenses.
 
+## Configuration Parameters
+
+### dataset_id
+
+The `dataset_id` is the ID of your BigQuery dataset where your billing data is stored. This is a unique identifier for the dataset within the project. You can find this ID in your Google Cloud Console under BigQuery.
```
[the remark this answers is not in the document]
Comment (shmsr) [reply packages/gcp/_dev/build/docs/billing.md:19]: I think we should add these two bullets somewhere in the doc:

```
- Ensure that you have the necessary permissions to access the dataset.
- The dataset should contain the billing export tables provided by GCP.
```

**Review by shmsr — CHANGES_REQUESTED** [review]

File: packages/gcp/docs/billing.md [hunk 6]
```diff
@@ -12,6 +12,41 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
 
 For detailed usage cost data, set the table pattern to `gcp_billing_export_resource_v1`. Detailed tables include the standard fields and additional fields, such as `effective_price`, enabling a more granular view of expenses.
```
Comment (ishleenk17) [inline packages/gcp/docs/billing.md:14]: @gpop63 : Let's specify the permissions required in GCP Cloud for this integration to fetch data appropriately.

**Review by Linu-Elias — APPROVED** [review]

**Review by shmsr — APPROVED** [review]

File: packages/gcp/_dev/build/docs/billing.md [hunk 7]
```diff
@@ -12,6 +12,57 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
...
+
+You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
+You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.
+
+Before using any GCP integration you will need:
+
+* **GCP Credentials** to connect with your GCP account.
+* **GCP Permissions** to make sure the service account you're using to connect has permission to share the relevant data.
+
+To collect GCP Billing metrics, you would need specific permissions to access the necessary data. Here's a list of permissions required:
```
Comment (alaudazzi) [inline packages/gcp/_dev/build/docs/billing.md:25]:
```suggestion
To collect GCP Billing metrics, the following permissions are required to access the necessary data:
```

File: packages/gcp/docs/billing.md [hunk 8]
```diff
@@ -12,6 +12,57 @@ For standard usage cost data, set the table pattern format to `gcp_billing_expor
...
+
+You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
+You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.
+
+Before using any GCP integration you will need:
+
+* **GCP Credentials** to connect with your GCP account.
+* **GCP Permissions** to make sure the service account you're using to connect has permission to share the relevant data.
+
+To collect GCP Billing metrics, you would need specific permissions to access the necessary data. Here's a list of permissions required:
```
Comment (alaudazzi) [inline packages/gcp/docs/billing.md:25]:
```suggestion
To collect GCP Billing metrics, the following permissions are required to access the necessary data:

```

**Review by alaudazzi — APPROVED** [review]
Left a few minor editing suggestions, otherwise LGTM.

**Pull request merged**